api/src/main/java/io/opencensus/metrics/MetricOptions.java

```diff
@@ -0,0 +1,157 @@
+  abstract Map<LabelKey, LabelValue> getConstantLabels();
+
+  /**
+   * Returns a new {@link Builder} with default options.
+   *
+   * @return a new {@code Builder} with default options.
+   * @since 0.5
+   */
+  public static Builder builder() {
+    return new AutoValue_MetricOptions.Builder()
+        .setDescription("")
+        .setUnit("1")
+        .setLabelKeys(Collections.<LabelKey>emptyList())
+        .setConstantLabels(Collections.<LabelKey, LabelValue>emptyMap());
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Sets the description of the Metric.
+     *
+     * @param description the description of the Metric.
+     * @return this.
+     */
+    public abstract Builder setDescription(String description);
+
+    /**
+     * Sets the unit of the Metric.
+     *
+     * @param unit the unit of the Metric.
+     * @return this.
+     */
+    public abstract Builder setUnit(String unit);
+
+    /**
+     * Sets the list of label keys for the Metric.
+     *
+     * @param labelKeys the list of label keys for the Metric.
+     * @return this.
+     */
+    public abstract Builder setLabelKeys(List<LabelKey> labelKeys);
+
+    /**
+     * Sets the map of constant labels (they will be added to all the TimeSeries) for the Metric.
+     *
+     * @param constantLabels the map of constant labels
+     * @return this.
+     */
+    // TODO: add support for this and make it public.
+    abstract Builder setConstantLabels(Map<LabelKey, LabelValue> constantLabels);
+
+    abstract MetricOptions autoBuild();
+
+    /**
+     * Builds and returns a {@code MetricOptions} with the desired options.
+     *
+     * @return a {@code MetricOptions} with the desired options.
+     * @since 0.20
+     * @throws NullPointerException if {@code description}, OR {@code unit} is null, OR {@code
+     *     labelKeys} is null OR any element of {@code labelKeys} is null, OR OR {@code
+     *     constantLabels} is null OR any element of {@code constantLabels} is null.
+     * @throws IllegalArgumentException if any {@code LabelKey} from the {@code labelKeys} is in the
+     *     {@code constantLabels}.
+     */
+    public MetricOptions build() {
+      MetricOptions options = autoBuild();
+      Utils.checkNotNull(options.getDescription(), "description");
+      Utils.checkNotNull(options.getUnit(), "unit");
+      Utils.checkListElementNotNull(
+          Utils.checkNotNull(options.getLabelKeys(), "labelKeys"), "labelKeys elements");
+      Utils.checkMapElementNotNull(
+          Utils.checkNotNull(options.getConstantLabels(), "constantLabels"),
+          "constantLabels elements");
+      Map<LabelKey, LabelValue> constantLabels = options.getConstantLabels();
+      for (LabelKey labelKey : options.getLabelKeys()) {
+        if (constantLabels.containsKey(labelKey)) {
+          throw new IllegalArgumentException("LabelKey in both labelKeys and constantLabels");
+        }
+      }
```

Comment:
is there a possibility of adding same key in LabelKeys? If so, should that be checked here as well?

Reply:
Added a check. even in the map may be possible because we need unique LabelKey.getKey()